docs/proposals/koordlet/20250425-heterogeneous-GPU-device-reporting.md

```diff
@@ -0,0 +1,221 @@
+                  "minors": [
+                      0
+                  ],
+                  "gpuLinkType": "NVLink",
+                  "ringBusBandwidth": "400Gi",
+                  "allocationScore": "1"
+                }
+        ],
+        "4": [
+                {
+                    "minors": [
+                       0,1,2,3 
+                    ],
+                    # This is the topology of the 910B.
+                    # https://www.hiascend.com/doc_center/source/zh/mindx-dl/50rc1/ref/dl_affinity_0002.html
+                    "gpuLinkType": "HCCS",
+                    "allocationScore": "1"
+                }
+        ]
+      }
+  name: node-1
+spec:
+  devices:
+  - health: true
+    id: GPU-fd971b33-4891-fd2e-ed42-ce6adf324615
+    minor: 0
+    resources:
+      koordinator.sh/gpu-core: "100"
+      koordinator.sh/gpu-memory: 15Gi
+      koordinator.sh/gpu-memory-ratio: "100"
+    topology:
+      busID: 0000:3b:00.0
+      nodeID: 0
+      pcieID: pci0000:3a
+      socketID: -1
+    type: gpu
+    # Add Healthy condition
+    conditions:
+    - lastHeartbeatTime: "2025-04-25T10:00:00Z"
+      lastTransitionTime: "2025-04-25T10:00:00Z"
+      status: "True"
+      type: Healthy
+  - health: true
+    id: GPU-0ca35a5f-551e-91e4-f3a3-c9b103afe1db
+    minor: 1
+    resources:
+      koordinator.sh/gpu-core: "100"
+      koordinator.sh/gpu-memory: 15Gi
+      koordinator.sh/gpu-memory-ratio: "100"
+    topology:
+      busID: 0000:3b:01.0
+      nodeID: 0
+      pcieID: pci0000:3a
+      socketID: -1
+    type: gpu
+    # Add Healthy condition
+    conditions:
+    - lastHeartbeatTime: "2025-04-25T10:00:00Z"
+      lastTransitionTime: "2025-04-25T10:00:00Z"
+      message: "XID 74: Nvlink Error"
+      reason: "NvlinkError"
+      status: "False"
+      type: Healthy
+
+```
+
+- Add the featuregate to enable the heterogeneous GPU device reporting. It is not recommended to enable both feature gates at the same time.
+```GO
+// NVIDIA Accelerators enables GPU related feature in koordlet. Only Nvidia GPUs supported.
+    Accelerators featuregate.Feature = "NvidiaAccelerator"
+// Accelerators enables heterogeneous GPU devices reporting.
+    Accelerators featuregate.Feature = "Accelerators"
+``` 
+
+### Device Infos Formate In File
+```Go
+type DeviceInfo struct {
+    Vendor string `json:"vendor"`
+    Model string `json:"model"` 
+    UUID string `json:"uuid"`//the Identifier of the device
+    Minor string `json:"minor"`// /dev/xxx0
+    Resources map[string]string `json:"resourece,omitempty"`
+    Topology *DeviceTopology `json:"topology,omitempty"`
+    Status *DeviceStatus `json:"status,omitempty"`
+}
+
+type DeviceTopology struct {
+    P2PLinks []DeviceP2PLink `json:"p2pLinks,omitempty"`
+}
+
+type DeviceP2PLink struct {
+	PeerMinor  string `json:"peerMinor"`
+	Type DeviceP2PLinkType `json:"type"`
+}
+
+type DeviceP2PLinkType string //like nvlink/hccs
+
+type DeviceStatus {
+    Healthy bool `json:"healthy"`
```

Comment:
Add a `LastCheckTime metav1.Time` to provide the `lastHeartbeatTime` of the device condition?